src/Interpreters/Session.cpp

```diff
@@ -0,0 +1,164 @@
+#include <Interpreters/Session.h>
+
+#include <Interpreters/Context.h>
+#include <Interpreters/SessionLog.h>
+#include <Access/ContextAccess.h>
+#include <Access/Credentials.h>
+#include <Access/AccessControlManager.h>
+#include <Access/User.h>
+
+namespace DB
+{
+
+Session::Session(const ContextPtr & context_to_copy, ClientInfo::Interface interface, std::optional<String> default_format)
```

Comment:
What is `context_to_copy`? Is this the global context?

Reply:
That depends, usually it is a server context or something similar.

Reply:
Added a comment to clarify things there